src/com/puppetlabs/mq.clj

```diff
@@ -132,10 +132,16 @@
     (start-broker! (build-embedded-broker brokername dir config))
     (catch java.io.EOFException e
       (log/warn
-        (str "Caught EOFException on broker startup, trying to restart it "
-             "again to see if that solves it. This is probably due to "
-             "KahaDB corruption."))
-      (start-broker! (build-embedded-broker brokername dir config)))))
+       "Caught EOFException on broker startup, trying again."
+       "This is probably due to KahaDB corruption "
+       "(see \"KahaDB Corruption\" in the PuppetDB manual).")
+      (start-broker! (build-embedded-broker brokername dir config)))
+    (catch java.io.IOException e
+      (log/error
+       e "Unable to start broker in " (pr-str dir) "."
+       "This is probably due to KahaDB corruption or version downgrade "
```

Comment:
I guess the wording in my mind was something like "or due to an incompatible KahaDB database after a downgrade.". Its usually corruption, but its sometimes this, so its worth saying both (or none, and sending them to the docs) otherwise people will always think it's corruption.

I also don't think mentioning the issue only in upgrade notes is enough, we should attach it to the existing KahaDB troubleshooting guide.